numba_dppy/tests/test_dpnp_functions.py

```diff
@@ -1411,5 +1411,88 @@ def f(a, b):
         self.assertTrue(max_abs_err < 1e-4)
 
 
+@unittest.skipUnless(
+    ensure_dpnp() and dpctl.has_gpu_queues(), "test only when dpNP and GPU is available"
+)
+class Testdpnp_array_ops_functions(unittest.TestCase):
+    tys = [np.int32, np.uint32, np.int64, np.uint64, np.float, np.double]
+
+    def check_take_for_different_datatypes(
+        self, fn, test_fn, ind, dims, tys, matrix=None
+    ):
+        for ty in tys:
+            if matrix:
+                a = np.arange(np.prod(np.array(dims)), dtype=ty).reshape(
+                    dims[0], dims[1]
+                )
+            else:
+                a = np.arange(dims[0], dtype=ty)
+
+            with dpctl.device_context("opencl:gpu"):
+                c = fn(a, ind)
+
+            d = test_fn(a, ind)
+            if c.shape == d.shape:
+                max_abs_err = np.all(c - d)
+            if not (max_abs_err < 1e-4) and c.dtype != d.dtype:
+                return False
+
+        return True
+
+    def get_np_array(self, a):
+        type_a = type(a)
+        if type_a is np.ndarray:
+            return a
+        elif type_a in (list, tuple):
+            return np.array(a)
+        else:
+            raise TypeError
+
+    def test_take_types(self):
+        @njit
+        def f(a, ind):
+            c = np.take(a, ind)
+            return c
+
+        test_indices = []
+        test_indices.append(np.array([1, 5, 1, 11, 3]))
+
+        with assert_dpnp_implementaion():
+            for ind in test_indices:
+                self.assertTrue(
+                    self.check_take_for_different_datatypes(
+                        f, np.take, self.get_np_array(ind), [12], self.tys, matrix=False
```

Comment:
I think you can just cast value to np.array here for indexes. Function `get_np_array` is redundant

Reply:
I also think the get_np_array is redundant, what are we trying to test throught this function?

Reply:
@reazulhoque 
The numpy.take function can accept indexes of the list and tuple types. We need to get the shape from the indexes to set the shape of the out array.
The get_np_array function allows us to pass indexes of the list and tuple types.